source/common/network/connection_impl.h

```diff
@@ -113,7 +113,11 @@ class ConnectionImpl : public FilterManagerConnection,
   void raiseEvent(ConnectionEvent event) override;
   // Should the read buffer be drained?
   bool shouldDrainReadBuffer() override {
-    return read_buffer_limit_ > 0 && read_buffer_.length() >= read_buffer_limit_;
+    if (read_buffer_limit_ > 0 && read_buffer_.length() >= read_buffer_limit_) {
+      ENVOY_CONN_LOG(warn, "read buffer size exceeded limit", *this);
```

Comment:
Can you comment a bit on the case you're trying to debug?

What this code does is say if the kernel has 500M in the kernel buffer, we won't read it all in one shot.  We read up to the read limit, and then cede, using setReadBufferReady to resume.

This is completely normal behavior and we definitely don't want to log at warning levels.  I don't think it's even advisable to log at trace level, but maybe if I understood the problem you're trying to solve I would understand the value of trace logs.

Reply:
Ah sorry, I just dug up the bug in question.  I think you're logging in the wrong spot.  Let me find some code pointers for you :-)

Reply:
OK, comment on the tracking issue since the details of what we have today may be helpful there.   As said over there I think we could add logs where we reset the stream, we could maybe add more helpful information to where we explain the details of sendLocalReply if the current explanations were not helpful, and if there's other useful docs (enhancing the docs on stats for example) that'd be great too!